Sources/HTTP/SimpleSocket/SimpleSocketConnectionListener.swift

```diff
@@ -100,29 +94,33 @@ public class BlueSocketConnectionListener: ParserConnecting {
             return
         }
         if (self.socket?.socketfd ?? -1) > 0 {
-            self.socket?.close()
+            self.socket?.shutdownAndClose()
         }
 
         //In a perfect world, we wouldn't have to clean this all up explicitly,
         // but KDE/heaptrack informs us we're in far from a perfect world
 
         if !(self.readerSource?.isCancelled ?? true) {
-            self.readerSource?.cancel()
+            #if os(Linux)
```

Comment:
This is mysterious... Why all the `#if`? Needs a comment at least.

Reply:
I'l comment it.  Basically, macOS >= 10.12 needs one implementation, and everything else needs a different one, but I couldn't figure out how to combine the `#if os()` check with the `#available` one.